org.lflang.ui/src/org/lflang/ui/LFUiModuleImpl.java

```diff
@@ -0,0 +1,317 @@
+                        IRegion candidate = strategy.findStartTerminal(document, command.offset);
+                        if (candidate != null) {
+                            if (bestStart == null || bestStart.getOffset() < candidate.getOffset()) {
+                                bestStrategy = strategy;
+                                bestStart = candidate;
+                            }
+                        }
+                    }
+                    if (bestStrategy != null) {
+                        bestStrategy.internalCustomizeDocumentCommand(document, command);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Handle combinations of nested braces.
+         * The following from the base class completely messes up with codeblocks.
+         * So we replace it below. Unfortunately, the base class CompoundMultiLineTerminalsEditStrategy
+         * is not written in a way that can be overridden, so we have to completely
+         * reimplement it.
+         */
+        @Override
+        public void configureCompoundBracesBlocks(final AbstractEditStrategyProvider.IEditStrategyAcceptor acceptor) {
+            acceptor.accept(new CompoundLFMultiLineTerminalsEditStrategy("{=", "=}"), IDocument.DEFAULT_CONTENT_TYPE);
+        }
+
+        /**
+         * For the given document, return the indentation of the line at
+         * the specified offset. If the indentation is accomplished with
+         * tabs, count each tab as four spaces.
+         * @param document The document.
+         * @param offset The offset.
+         */
+        public static int indentationAt(final IDocument document, final int offset) {
+            try {
+                final int lineNumber = document.getLineOfOffset(offset); // Line number.
+                final int lineStart = document.getLineOffset(lineNumber); // Offset of start of line.
+                final int lineLength = document.getLineLength(lineNumber); // Length of the line.
+                String line = document.get(lineStart, lineLength);
+                // Replace all tabs with four spaces.
+                line = line.replaceAll("\t", "    ");
+                return line.indexOf(line.trim());
+            } catch (Throwable e) {
+                throw Exceptions.sneakyThrow(e);
+            }
+        }
+
+        /**
+         * When encountering {= append =}.
+         */
+        protected void configureCodeBlock(final AbstractEditStrategyProvider.IEditStrategyAcceptor acceptor) {
+            acceptor.accept(new SingleLineTerminalsStrategy("{=", "=}", SingleLineTerminalsStrategy.DEFAULT) {
+                                @Override
+                                public void handleInsertLeftTerminal(final IDocument document,
+                                                                     final DocumentCommand command) throws BadLocationException {
```

Comment:
This line is 129 characters long. Maybe this entire block (for `SingleLineTerminalsStrategy`) does not need this much indentation?

Reply:
Removed quite some unnecessary indentations in https://github.com/lf-lang/lingua-franca/pull/853/commits/5740fc5b368cceac825f52180b70a1ff15c6baab.